src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesVerifiers.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.jenkins.plugins.k8sengine;
+
+import com.google.common.base.Preconditions;
+import com.jayway.jsonpath.JsonPath;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+/**
+ * Verification adapters for verifying ManifestObjects using kubectl and returning and logging the
+ * result. Inner class, {@link VerificationResult} encapsulates the result and the class will keep a
+ * registry of Verifiers per Kubernetes API Kind and Version.
+ */
+public class KubernetesVerifiers {
+  private static final Logger LOGGER = Logger.getLogger(KubernetesEngineBuilder.class.getName());
+  private static Map<String, Verifier> verifiers = new HashMap<String, Verifier>();
+
+  private static final Verifier defaultVerifier = new DefaultVerifier();
+
+  public static final String DEPLOYMENT_KIND = "deployment";
+
+  // Register the available verifiers.
+  static {
+    verifiers.put("*/deployment", new DeploymentVerifier());
+  }
+
+  /**
+   * Represents the result object for verification action. The status field represents whether the
+   * verification succeeded. The toString() returns relevant information pertaining to why the
+   * status is what it is.
+   */
+  public static class VerificationResult {
+    private String log;
+    private boolean status;
+    private Manifests.ManifestObject manifestObject;
+
+    /** @return information relevant to the result and why the status is what it is. */
+    private String getLog() {
+      return log;
+    }
+
+    /** @return if the Kubernetes object was verified. */
+    public boolean isVerified() {
+      return status;
+    }
+
+    /** @return the {@link Manifests.ManifestObject} for which verification was attempted. */
+    public Manifests.ManifestObject getManifestObject() {
+      return manifestObject;
+    }
+
+    /**
+     * A class that encapsulates information about the verification result including whether it was
```

Comment:
No need to duplicate the class header, "Constructs a new {@link VerificationResult}." suffices.